File: examples/log_analyzer.clj
(ns log-analyzer)

;; OpenClaw log analyzer (Clojure) — parses gateway access logs from stdin
(def pattern #"^(\S+)\s+(INFO|WARN|ERROR)\s+(\S+)\s+(.+)$")

(defn -main [& _]
  (let [counts (atom {"INFO" 0 "WARN" 0 "ERROR" 0})]
    (doseq [line (line-seq (java.io.BufferedReader. *in*))]
      (when-let [[_ ts level node msg] (re-matches pattern line)]
        (swap! counts update level inc)
        (when (= level "ERROR")
          (println (str "⚠ " ts " [" node "] " msg)))))
    (println "\n--- Summary ---")
    (doseq [level ["ERROR" "INFO" "WARN"]]
      (println (str level ": " (@counts level))))))

(-main)

File: examples/LogAnalyzer.kt
// OpenClaw log analyzer (Kotlin) — parses gateway access logs from stdin
fun main() {
    val pattern = Regex("""^(\S+)\s+(INFO|WARN|ERROR)\s+(\S+)\s+(.+)$""")
    val counts = mutableMapOf("INFO" to 0, "WARN" to 0, "ERROR" to 0)

    generateSequence(::readLine).forEach { line ->
        val m = pattern.matchEntire(line) ?: return@forEach
        val (ts, level, node, msg) = m.destructured
        counts[level] = counts.getValue(level) + 1
        if (level == "ERROR") println("⚠ $ts [$node] $msg")
    }

    println("\n--- Summary ---")
    for (level in listOf("ERROR", "INFO", "WARN")) println("$level: ${counts[level]}")
}
